### PR TITLE
New data source: `azurerm_monitor_workspace`

### LINKED ISSUE #23905: azurerm_monitor_workspace data source
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Feature request for Azure monitor workspace data source. As far as I know, there is no data source reflecting the terraform resource `azurerm_monitor_workspace`.

### New or Affected Resource(s)/Data Source(s)

data source: azurerm_monitor_workspace

### Potential Terraform Configuration

```hcl
data "azurerm_monitor_workspace" "example" {
  name                = "example"
  resource_group_name = azurerm_resource_group.example.name
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by favoretti**
Fixes #23905

```
$ TF_ACC=1 go test -v ./internal/services/monitor -timeout=1000m -run='TestAccMonitorWorkspaceDataSourceDataSource_complete'
=== RUN   TestAccMonitorWorkspaceDataSourceDataSource_complete
=== PAUSE TestAccMonitorWorkspaceDataSourceDataSource_complete
=== CONT  TestAccMonitorWorkspaceDataSourceDataSource_complete
--- PASS: TestAccMonitorWorkspaceDataSourceDataSource_complete (164.92s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/monitor	166.554s
```

### CODE REVIEW

**Comment by favoretti** [comment]
image

File: internal/services/monitor/monitor_workspace_data_source.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type WorkspaceDataSource struct{}
+
+var _ sdk.DataSource = WorkspaceDataSource{}
+
+func (d WorkspaceDataSource) ModelObject() interface{} {
+	return &WorkspaceDataSource{}
+}
```
Comment (stephybun) [inline internal/services/monitor/monitor_workspace_data_source.go:27]: We should define the data source model here.

File: internal/services/monitor/monitor_workspace_data_source.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+func (d WorkspaceDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Monitor.WorkspacesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state WorkspaceResourceModel
```
Comment (stephybun) [inline internal/services/monitor/monitor_workspace_data_source.go:69]: Although the resource and data source models might be identical now they could drift over time, so we should define a data source model above and use that here.

File: website/docs/d/monitor_workspace.html.markdown [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+* `id` - The ID of the Resource.
+
+* `kind` - The kind of the Workspace. Possible values are `Linux` and `Windows`.
+
+* `location` - The Azure Region where the Workspace is located.
+
+* `query_endpoint` - The query endpoint for the Azure Monitor Workspace.
+
+* `public_network_access_enabled` - Whether network access from public internet to the Workspace are allowed. Possible values are `true` and `false`.
```
Comment (stephybun) [inline website/docs/d/monitor_workspace.html.markdown:43]:
```suggestion
* `public_network_access_enabled` - Whether network access from public internet to the Workspace are allowed.
```

File: website/docs/d/monitor_workspace.html.markdown [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+* `tags` - A mapping of tags which should be assigned to the Data Collection Endpoint.
```
Comment (stephybun) [inline website/docs/d/monitor_workspace.html.markdown:45]:
```suggestion
* `tags` - A mapping of tags that are assigned to the Workspace.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @favoretti.

Overall this looks good, but we should define a data source model and use that when encoding/decoding. Once that's done this should be good to go!

**Review by stephybun — APPROVED** [review]
Thanks @favoretti LGTM 👍

**Pull request merged**